zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/distribution/DbDistributionState.java

```diff
@@ -117,40 +134,54 @@ public void addCommandDistribution(
     this.distributionKey.wrapLong(distributionKey);
     commandDistributionRecordColumnFamily.insert(
         this.distributionKey, new PersistedCommandDistribution().wrap(commandDistributionRecord));
+
+    getMetrics().addDistribution(distributionKey);
   }
 
   @Override
   public void removeCommandDistribution(final long distributionKey) {
     this.distributionKey.wrapLong(distributionKey);
     commandDistributionRecordColumnFamily.deleteIfExists(this.distributionKey);
+
+    getMetrics().removeDistribution(distributionKey);
   }
 
   @Override
   public void addRetriableDistribution(final long distributionKey, final int partition) {
     this.distributionKey.wrapLong(distributionKey);
     partitionKey.wrapInt(partition);
     retriableDistributionColumnFamily.insert(distributionPartitionKey, DbNil.INSTANCE);
+
+    getMetrics().addInflightDistribution(partition, distributionKey);
   }
 
   @Override
   public void removeRetriableDistribution(final long distributionKey, final int partition) {
     this.distributionKey.wrapLong(distributionKey);
     partitionKey.wrapInt(partition);
     retriableDistributionColumnFamily.deleteExisting(distributionPartitionKey);
+
+    getMetrics().removeInflightDistribution(partition, distributionKey);
   }
 
   @Override
   public void addPendingDistribution(final long distributionKey, final int partition) {
     this.distributionKey.wrapLong(distributionKey);
     partitionKey.wrapInt(partition);
     pendingDistributionColumnFamily.upsert(distributionPartitionKey, DbNil.INSTANCE);
+
+    // TODO: the upsert above makes counting metrics difficult as this is currently called twice
+    // (ENQUEUED/DISTRIBUTING) and therefore would also be counted twice.
+    getMetrics().addPendingDistribution(partition, distributionKey);
```

Comment:
here we need to find a solution to prevent counting the `add` twice, e.g. by clearly distinguishing between queue vs no-queue strategy.

Reply:
We could look into removing the duplicate `addPendingDistribution` call. I can't recall why have this.

Reply:
Yup, still I wanted to think about how to make this more robust, thus it's harder to break this again by accident. e.g. is there a way to know from the `upsert` if it was an insert or delete 🤔